src/components/Settings/ChangePassword.vue

```diff
@@ -59,8 +61,16 @@ export default {
   methods: {
     save () {
       const { oldPassword, newPassword } = this
-      this.$emit('save', { oldPassword, newPassword })
```

Comment:
This new implementation of the `save` method no longer triggers an event for the parent component to call the backend. The reason is that this doesn't work in combination with a Promise.
As a result, the `changePassword` method of the `Settings` component (the parent component) is no longer used.
Do you consider this bad style? Is there a nicer way of clearing the password fields?

Reply:
It's not the style we've used so far, which would be to set some state (which might be status), but I don't feel like we've quite got all the common use cases covered - it seems very long winded for cases like this, and not really useful. It might be there is a reasonable similar approach that has that logic in a connect component, I don't think we've explored all the different options that those take.

Although maybe there is not time now to investigate further approaches, and that this is fine for now.

One thing though, it should use the async/await syntax which is equivelent to using the promises explicitly. For that you need to declare the whole `save()` function async (`async save()`) then just await on `this.saveUser()` (if there is an error it should throw, so should not need to check the result.

Reply:
The vuejs guy points out you can emit a callback in the event too... https://github.com/vuejs/vue/issues/5443#issuecomment-294147285

Reply:
We already pass actions as props in some other places (e.g. https://github.com/yunity/karrot-frontend/blob/f7377683baa1a4cbc1a328278f691f6e271c1499/src/components/Conversation/Conversation.vue#L120 ), so it seems fine to do the same here. Emitting a callback could be easier to understand though.

Reply:
Agree with switching `.then` to `await`, we almost always use `async`/`await` in our codebase (could refactor the rest at some point...)

Reply:
Aah, emitting a callback - that's clever. 
I implemented it this way now (6fd91c3).